Sources/Bot/Services/GitHubService.swift

```diff
@@ -43,10 +50,37 @@ public final class GitHubService {
             return SignalProducer(error: .unknown)
         }
     }
+
+    private static func signatureVerifier(
+    ) -> (Token) -> (RequestProtocol) -> Result<RequestProtocol, EventHandlingError> {
```

Comment:
Why 3 levels of currying instead of two? Wouldn't this be enough?

```swift
private static func signatureVerifier(token: Token) -> (RequestProtocol) -> Result<RequestProtocol, EventHandlingError>
```

_And then in the `init` you could avoid the awkward `()(signatureToken)` syntax and just do this:_
```swift
self.signatureVerifier = GitHubService.signatureVerifier(signatureToken)
```

Reply:
This makes no sense, you're right.